package.json

```
@@ -2,65 +2,78 @@
   "name": "fast-chunk-string",
   "version": "1.0.1",
   "description": "Chunk string into equal substrings with unicode support",
-  "license": "MIT",
+  "keywords": [
+    "string",
+    "split",
+    "chunk",
+    "unicode",
+    "utf",
+    "utf8",
+    "substring"
+  ],
   "repository": "vladgolubev/fast-chunk-string",
+  "license": "MIT",
   "author": {
     "name": "Vlad Holubiev",
     "email": "golubev.vld@gmail.com",
     "url": "vladholubiev.com"
   },
-  "engines": {
-    "node": ">=6"
-  },
+  "files": [
+    "src/index.ts"
+  ],
   "scripts": {
     "benchmark": "node benchmark",
+    "build": "rm -rf lib/ && yarn build:types && babel src --out-dir lib --ignore '**/*.test.ts' --extensions '.ts'",
+    "build:types": "tsc --emitDeclarationOnly --declaration --isolatedModules false --declarationDir lib",
     "coverage": "jest --coverage",
-    "lint": "eslint . --fix",
-    "test": "jest"
+    "lint": "eslint . --ext .js,.ts,.json --fix",
+    "lint:ci": "eslint . --ext .js,.json,.ts --quiet",
+    "prepack": "yarn build",
+    "test": "jest src",
+    "type-check": "tsc --noEmit",
+    "type-check:watch": "npm run type-check -- --watch"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
+      "post-commit": "git update-index --again"
+    }
+  },
+  "lint-staged": {
+    "*.{html,json,md,yml}": [
+      "prettier --write --ignore-path=./.eslintignore",
+      "git add"
+    ],
+    "*.{ts,js}": [
+      "eslint --fix",
+      "git add"
+    ]
+  },
+  "jest": {
+    "testEnvironment": "node"
   },
-  "files": [
-    "index.js"
-  ],
-  "keywords": [
-    "string",
-    "split",
-    "chunk",
-    "unicode",
-    "utf",
-    "utf8",
-    "substring"
-  ],
   "dependencies": {
     "runes": "0.4.3",
     "string-length": "3.1.0"
   },
   "devDependencies": {
+    "@babel/cli": "^7.8.0",
+    "@babel/core": "^7.8.0",
+    "@babel/preset-env": "^7.8.2",
+    "@babel/preset-typescript": "^7.8.0",
     "@shelf/eslint-config": "0.13.0",
+    "@types/jest": "^24.0.25",
+    "@types/node": "^13.1.6",
```

```suggestion
    "@types/node": "12",
```

Since this is the target version in your `.babelrc.js`